modules/GenericDataParser/src/client/GenericDataParser.coffee

```diff
@@ -8,3 +8,30 @@ class GenericDataParserController extends BasicFileValidateAndSaveController
 		@loadImagesFile = true
 		super()
 		@$('.bv_moduleTitle').html("Simple #{window.conf.experiment.label} Loader")
+
+	validateParseFile: =>
+		
+		# check the number of files uploaded by checking the number of rows in the table of uploaded files
+		# if it is more than one, we want to block the validation 
+		filesUploaded = @$(".files tr.template-download").length
```

Comment:
Seems like all of this logic should be handled down the LSFileChooserController where it [detects that it has reached the max number of files] https://github.com/mcneilco/acas/blob/a78c855f3793e981fb58ad4cfb17c9c954ae788f/modules/Components/src/client/LSFileChooser.coffee#L104

Reply:
@brianbolt I thought so too and actually started with that (and also the maxNumberOfFiles option) in the LSFileChooserController but didn't have any luck with toggling it. I observed some odd behavior that I couldn't figure out so opted for this current approach. Basically the currentNumberOfFiles variable which is compared against maxNumberOfFiles would go to zero every time, so setting the maxNumberOfFiles wouldn't matter.